### PR TITLE
Call Add shelf endpoint

### LINKED ISSUE #745: Call add shelf endpoint
**Describe the solution you'd like**
When we click on the [add shelf button](https://github.com/Project-Books/book-project/blob/main/frontend/src/my-books/MyBooks.tsx#L175), we should call the add shelf endpoint: https://github.com/Project-Books/book-project/blob/0.2.0/backend/src/main/java/com/karankumar/bookproject/backend/controller/ShelfController.java#L4.

- Verb: POST
- Endpoint: `localhost:8080/api/shelf`
- Pass in shelf name as a path variable

This endpoint call should be added here: https://github.com/Project-Books/book-project/blob/main/frontend/src/my-books/ShelfModal.tsx#L38

We would need to handle the bad requests that we get back accordingly with error messages (red text)

### PR DESCRIPTION
**Pull request by priya1712d**
## Summary of change

Added the code to call ADD SHELF endpoint

## Related issue

Closes #745

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

### CODE REVIEW

**Comment by knjk04** [comment]
Hi @priya1712d, thanks for creating this PR. The linter is failing -- could you try running `yarn install` and then commit the changes? Thanks

**Comment by knjk04** [comment]
@priya1712d The build is still failing. Could you please look into this?

**Comment by priya1712d** [comment]
@knjk04 Could you approve this one more time?

File: frontend/src/my-books/ShelfModal.tsx [hunk 1]
```diff
@@ -19,25 +19,46 @@ import React, { Component } from "react";
...
 
     submitShelf = (event: React.MouseEvent<HTMLButtonElement>): void => {
         event.preventDefault();
+        const shelfName = this.state.name;
+        if(shelfName.length > 0 && shelfName.toString().length > 20){
+            this.setState({showError: true, msg: "Shelf name is too long"});
+            return;
+        }
+        HttpClient.post(Endpoints.shelf, shelfName).then(() => {
+            this.setState({showError: false, showInfo: true, msg: "Shelf saved successfully"});
```
Comment (knjk04) [inline frontend/src/my-books/ShelfModal.tsx:55]: This was a nice touch!

**Review by knjk04 — APPROVED** [review]
Thanks for implementing this! 

I fixed the linter error by running `yarn install` and fixing the eslint warnings, but otherwise this was great!

**Pull request merged**